gems/aws-sdk-s3/sig/customizations/object.rbs

```diff
@@ -0,0 +1,38 @@
+module Aws
+  module S3
+    class Object
+      alias size content_length
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#copy_from-instance_method
+      def copy_from: (untyped source, ?Hash[Symbol, untyped] options) -> void
+                   | ...
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#copy_to-instance_method
+      def copy_to: (untyped target, ?Hash[Symbol, untyped] options) -> void
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#move_to-instance_method
+      def move_to: (untyped target, ?Hash[Symbol, untyped] options) -> void
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#presigned_post-instance_method
+      def presigned_post: (?Hash[Symbol, untyped]) -> untyped
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#presigned_url-instance_method
+      def presigned_url: ((:get | :put | :head | :delete | :create_multipart_upload | :list_multipart_uploads | :complete_multipart_upload | :abort_multipart_upload | :list_parts | :upload_part) method, ?Hash[Symbol, untyped] params) -> String
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#presigned_request-instance_method
+      def presigned_request: ((:get | :put | :head | :delete | :create_multipart_upload | :list_multipart_uploads | :complete_multipart_upload | :abort_multipart_upload | :list_parts | :upload_part) method, ?Hash[Symbol, untyped] params) -> [String, Hash[String, String]]
```

Comment:
I would have to check (on mobile right now) but I think the presigned methods can take any operation as a symbol. Does it make sense to be less exhaustive here?

Reply:
I followed the documentation and did not know that other values could be valid.
If you want to be more lax, we can use `String | Symbol` instead.

Reply:
Up to you, I think documenting Symbol makes sense. It can be any method.